File: src/it/java/cloud/cosmin/checklister/ListUpdateIT.java
package cloud.cosmin.checklister;

import cloud.cosmin.checklister.dto.ListGetDto;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ListUpdateIT extends BaseIT {
    @Test
    public void listUpdate() {
        ListGetDto newList = createList("testlist");
        ListGetDto updatedList = updateList(newList.id, "newtitle");
        assertEquals("newtitle", updatedList.title);
    }
}

File: checklister-web/src/integrationTest/kotlin/cloud/cosmin/checklister/ListUpdateIT.kt
package cloud.cosmin.checklister

import cloud.cosmin.checklister.lib.dto.ItemPostDto
import org.junit.Assert.assertEquals
import org.junit.Test

class ListUpdateIT : BaseIT() {
    @Test
    fun listUpdate() {
        val (id) = createList("testlist")
        val (_, title) = updateList(id!!, "newtitle")
        assertEquals("newtitle", title)
    }

    @Test
    fun moveListItem() {
        val (id) = createList("testlist")
        val itemPost = ItemPostDto(id, "content", "text/plain")
        val (itemId) = addItem(itemPost)

        val (id2) = createList("list2")
        val updatedItem = moveItem(id2!!, itemId!!)
        assertEquals(id2, updatedItem!!.list)
    }
}
